internal/lexer/lexer.go

```diff
@@ -80,7 +80,12 @@ func (l *Lexer) ConsumeInt() int {
 
 func (l *Lexer) ConsumeFloat() float64 {
 	text := l.sc.TokenText()
-	l.ConsumeToken(scanner.Int)
+	switch l.next {
```

Comment:
I think this should simply be `l.ConsumeToken(scanner.Float)`, since the token kind already gets checked before this method is called.